### PR TITLE
Add the --allow_different_domains flag

### PR DESCRIPTION
**Pull request by emilyalbini**
This new flag ignores the error raised when you change the list of SANs. That error is generally useful, because it alerts you of a mistake you might have made, for example forgetting to add a domain, but it can be annoying if you generate the list of domains from a script, and that list changes over time.

A possible use case for this is a list of domains managed by a configuration manager, and a script which generates simp_le's command line arguments and calls it. In that case, you're sure the list is correct, and maybe you're just adding a domain.

Since in that environments a manual fix can be annoying, the newly introduced flag disables that behavior if provided. This way, the error is kept for the common use cases, but if someone needs to ignore it he can.

Discussed in #36. In my particular case, I manage that file with ansible, and it really bugs me the fact I need to log into the server and remove the certificates manually.

### CODE REVIEW

**Comment by emilyalbini** [comment]
I had some troubles figuring out the name of the flag. If you've better ideas... :-)

File: simp_le.py [hunk 1]
```diff
@@ -850,6 +850,10 @@ def create_parser():
         '--reuse_key', action='store_true', default=False,
         help='Reuse private key if it was previously persisted.',
     )
+    io_group.add_argument(
+        '--renew_different_san', action='store_true', default=False,
+        help='Don\'t abort if the SANs changed'
```
Comment (Lekensteyn) [inline simp_le.py]: What about something like:
"Allow replacement of a certificates with different SANs" or
"Force certificate renewal when domains are modified"

That should make it clearer what happens instead. Some suggested option names: `--allow_different_san` or `--allow_domain_changes`

Comment (emilyalbini) [inline simp_le.py]: I think `--allow_domain_changes` is the most suitable of the proposed ones. @kuba?

**Comment by emilyalbini** [comment]
Ok, I've renamed the flag to `--allow_different_domains`.

File: simp_le.py [hunk 2]
```diff
@@ -881,6 +881,10 @@ def create_parser():
         '--reuse_key', action='store_true', default=False,
         help='Reuse private key if it was previously persisted.',
     )
+    io_group.add_argument(
+        '--allow_domain_changes', action='store_true', default=False,
+        help='Allow changes in the SANs of the certificate.'
```
Comment (kuba) [inline simp_le.py:886]: nit: missing comma at the end of the line

File: simp_le.py [hunk 3]
```diff
@@ -1384,7 +1395,8 @@ def main_with_exceptions(cli_args):
     check_plugins_persist_all(args.ioplugins)
 
     existing_data = load_existing_data(args.ioplugins)
-    if valid_existing_cert(existing_data.cert, args.vhosts, args.valid_min):
+    if valid_existing_cert(existing_data.cert, args.vhosts, args.valid_min,
+                           args.allow_domain_changes):
```
Comment (kuba) [inline simp_le.py:1399]: can you use `different_domains=args.allow_domain_changes`?

**Comment by kuba** [comment]
I haven't made my mind yet, but I think we're missing integration test here (nothing makes sure that the flag actually changes behavior of the client).

**Comment by emilyalbini** [comment]
Do you want to make this pull request the default behavior as discussed in IRC, or should I fix the things you pointed out?

**Comment by kuba** [comment]
I have a local branch to fix this problem. However, I'm offline for around a week.

**Comment by emilyalbini** [comment]
Ok, I'll close this pull request then.

**Comment by kuba** [comment]
Fixed in  e30ad8c.

**Pull request closed without merging**